### PR TITLE
Hotfix/rm node config

### LINKED ISSUE #97: Remove node-config
Node config is great for a full application, but this stand-alone module can function outside of the main `Punchcard` application. It should not be required to use node-config and should accept configuration as a parameter.

```gherkin
Feature: Config as param
  As an implementor
  I want to directly control content-type's config
  So that I can dynamically configure it based on multiple config sources

  Scenario: Custom Config
    Given that I provide core configuration for `content-types`
      And that my users may optionally provide my application with their own configuration
      And that I want to control how user-provided and my core configurations are merged
     When I create content types or forms
     Then they should be generated based on the provided configuration that I control
      And it should also check the existing defaults
      And if no configuration is provided the existing defaults should be used
```

---

**Config as param**
- [ ] Custom Config

### PR DESCRIPTION
**Pull request by scottnath**
Content-types now gets it's configuration as a parameter instead of using `node-config`

---
Resolves #97 

`DCO 1.1 Signed-off-by: Scott Nath <github@scottnath.com>`

### CODE REVIEW

File: lib/content-types.js [hunk 1]
```diff
@@ -36,40 +36,51 @@ const raw = loader;
...
+ *
+ * @param  {string} type - id of a content type
+ * @param  {object} values - contains values for inputs
+ * @param  {array} loadedTypes - array of merged content type objects
+ * @param  {object} config - configuration object
+ *
+ * @returns {promise} resolves with promise from `only` function
+ */
+const onlyCT = (type, values, loadedTypes, config) => {
+  const userConfig = values || {};
```
Comment (Snugug) [inline lib/content-types.js]: I would expect `userConfig` to be a version of `config` not `values`; can this be something like `overrides`

File: lib/content-types.js [hunk 2]
```diff
@@ -36,40 +36,51 @@ const raw = loader;
...
 
   if (loadedTypes) {
     const ct = findCT(type, onlyTypes);
 
-    return only(ct, userConfig);
+    return only(ct, userConfig, config);
   }
 
-  return merged().then(types => {
+  return merged('', config).then(types => {
```
Comment (Snugug) [inline lib/content-types.js]: Why are you passing an empty string in? Should you pass something like `null`, or an empty object?

File: lib/content-types/load.js [hunk 3]
```diff
@@ -3,28 +3,33 @@
...
 
 /**
  * Gets all Content Types
  *
+ * @param {object} configuration - configuration object
+ *
  * @returns {Promise} - An {Array} containing {Object} for each content type, including `name` {string}, `id` name {string}, and full config {object} loaded from YAML config file
  */
-const load = () => {
+const load = (configuration) => {
```
Comment (Snugug) [inline lib/content-types/load.js]: Can we keep the parameter name consistent throughout the project for configuration?

File: lib/content-types/load.js [hunk 4]
```diff
@@ -33,50 +38,50 @@ const load = () => {
...
       // find attribute which is the identifier
-      const attr = config.attributes.find(a => {
-        return a.id === config.identifier;
+      const attr = type.attributes.find(a => {
+        return a.id === type.identifier;
       });
 
-      if (!attr) {
-        reject(new Error(`Identifier '${config.identifier}' is not an attribute in content type '${config.name}'.`));
+      if (attr === undefined) {
```
Comment (Snugug) [inline lib/content-types/load.js]: `typeof attr === 'undefined'`

File: lib/content-types.js (hunk 1 above)
Comment (scottnath) [inline lib/content-types.js]: Changed to:
```
const onlyCT = (type, vals, loadedTypes, config) => {
  const values = vals || {};
```

File: lib/content-types.js (hunk 2 above)
Comment (scottnath) [inline lib/content-types.js]: has to be `null`. change added.

File: lib/content-types/only.js [hunk 5]
```diff
@@ -3,11 +3,19 @@
...
-const only = (type, configuration) => {
+/**
+ * Retrieves one content type's config object, merged with input plugins and a form's values
+ *
+ * @param  {object} type merged content type object
+ * @param  {object} vals values for inputs
+ *
+ * @returns {object}  content type object merged with input plugins and values
+ */
+const only = (type, vals) => {
```
Comment (Snugug) [inline lib/content-types/only.js]: Again, I'd like consistent function signatures. `values` instead of `vals`

Comment (Snugug) [inline lib/content-types/only.js]: And in previous function this was called `userConfig`, and I'd still like it to be something like `overrides`

Comment (scottnath) [inline lib/content-types/only.js]: ✔️  changed to `overrides`

File: lib/content-types/load.js (hunk 4 above)
Comment (scottnath) [inline lib/content-types/load.js]: fixed

**Comment by scottnath** [comment]
@Snugug all changes implemented.

**Pull request merged**